### PR TITLE
pytorch 2.1.1 torchvision 0.16.1

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

- [x] `resource` blocks have been checked for updates.

### CODE REVIEW

File: Formula/p/pytorch.rb [hunk 1]
```diff
@@ -4,8 +4,8 @@ class Pytorch < Formula
   desc "Tensors and dynamic neural networks"
   homepage "https://pytorch.org/"
   url "https://github.com/pytorch/pytorch.git",
-      tag:      "v2.1.0",
-      revision: "7bcf7da3a268b435777fe87c7794c382f444e86d"
+      tag:      "2.1.1",
```
Comment (onlined) [inline Formula/p/pytorch.rb:7]: The correct tag is `2.1.1`.

```suggestion
      tag:      "v2.1.1",
```

Comment (ZhongRuoyu) [reply Formula/p/pytorch.rb]: Thanks

**Comment by ZhongRuoyu** [comment]
We should probably have a look at https://github.com/orgs/Homebrew/discussions/4916 too.

**Comment by singingwolfboy** [comment]
> It's very likely that `pytorch` is vendoring its own `fmt`, but that really shouldn't be allowed.

I just took a look at the Pytorch source tree, and it's doing exactly that. However, Pytorch vendors a LOT of its dependencies, and only provides options to use system deps instead for a few of them. There does not seem to be an option to tell Pytorch to use the system `fmt` library, unfortunately. I also checked the Pytorch issue tracker, and I don't see an issue about this.

**Comment by cho-m** [comment]
Kind of curious why our bottle has `fmt` headers installed while I didn't see them in other distributions (e.g. [Arch](https://archlinux.org/packages/extra/x86_64/python-pytorch/files/), [PyPI](https://pypi.org/project/torch/#files), FreeBSD). May be the exact build method.

**Comment by chenrui333** [comment]
yeah, they did the vendor out via submodule, I wonder if we should just do the `link_overwrite`

**Comment by cho-m** [comment]
I can confirm that `fmt` headers are not installed if you let upstream's build scripts handle CMake configuration. However, the directory structure will be set up inside site-packages.

- Letting `setup.py` handle configuration:
    ```console
    ❯ ls $(brew --prefix pytorch)/libexec/lib/python3.11/site-packages/torch/include -1
    ATen
    c10
    caffe2
    tensorpipe
    torch
    clog.h
    cpuinfo.h
    fp16.h
    fxdiv.h
    libshm.h
    psimd.h
    pthreadpool.h
    qnnpack_func.h
    sleef.h
    xnnpack.h
    ```
- Our formula bottle from directly running CMake:
    ```console
    ❯ ls $(brew --prefix pytorch)/include -1
    ATen
    c10
    caffe2
    fmt
    fp16
    kineto
    tensorpipe
    torch
    clog.h
    cpuinfo.h
    fp16.h
    fxdiv.h
    libshm.h
    psimd.h
    pthreadpool.h
    qnnpack_func.h
    sleef.h
    xnnpack.h
    ```

**Pull request closed without merging**